csrc/id_model/id_graph.cpp

```diff
@@ -0,0 +1,540 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <id_model/id_graph.h>
+#include <id_model/to_string.h>
+#include <ir/utils.h>
+
+namespace nvfuser {
+
+namespace {
+using UnorderedSetOfExprGroup = std::unordered_set<ExprGroup>;
```

Comment:
I don't mind calling it `SetOfExprGroup` to save some typing. `ExprGroup` isn't sortable anyway, so it's naturally an unordered set.

Reply:
I'd prefer to keep it as is. `SetOfExprGroup` to me implies `std::set`.